arrow/arrow.py

```diff
@@ -118,11 +132,19 @@ def utcfromtimestamp(cls, timestamp):
 
     @classmethod
     def fromdatetime(cls, dt, tzinfo=None):
-        ''' Constructs an :class:`Arrow <arrow.arrow.Arrow>` object from a ``datetime`` and optional
-        ``tzinfo`` object.
+        ''' Constructs an :class:`Arrow <arrow.arrow.Arrow>` object from a ``datetime`` and
+        optional replacement timezone.
 
         :param dt: the ``datetime``
-        :param tzinfo: (optional) a ``tzinfo`` object.  Defaults to UTC.
+        :param tzinfo: (optional) A :ref:`timezone expression <tz-expr>`.  Defaults to ``dt``'s
+            timezone, or UTC if naive.
+
+        If you only want to replace the timezone of naive datetimes::
+
+            >>> dt
+            datetime.datetime(2013, 5, 5, 0, 0, tzinfo=tzutc())
+            >>> arrow.Arrow.fromdatetime(dt, dt.tzinfo or 'US/Pacific')
+            <Arrow [2013-05-05T00:00:00+00:00]>
```

Comment:
I figured this would be a common enough need that it should be documented.  It also highlights that `tzinfo` _always_ replaces the timezone.